### DISCUSSION TITLE
error message 404

### DISCUSSION THREAD

**Discussion by Cawaale256** [community/community#132871] [Programming Help]
### Body

I am writing html and css and I am new to github .In particular I am uploading image on page and keeping recieving this message 404

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment]
Hi @Cawaale256 ,

I don't know what exactly your issue might be because you haven't given enough details, but I suspect it's a path problem, so I'm going to show some solutions to common file path errors that might help you with your problem.

Let's imagine that you have the following folder structure:

`/your-repo`
`├── index.html`           
`└── /assets`
         `        ├────/css`
         `        │`         `    └──── style.css`
         `        └────/img`
         `               └──── image.jpg`

The correct path in the `index.html` for your `css` and `img` files would be the following:

```html
<link rel="stylesheet" href="assets/css/style.css" type="text/css" />
<img src="assets/img/image.jpg">
```
You can also call your image file from `style.css`, but in this case you must use a relative path because the location of `image.jpg` is relative to the `style.css` directory:

```css
img {
    background: url("../img/image.jpg");
}
```

Absolute paths always start with the root directory and provide the full path to the file or directory, as in the first example, because the `index.html` is in the root.

On the other hand, a relative path is a path to a file or directory that is relative to the current directory as in the second example, in this case you have to use `../` followed by the directory containing the file.

**Comment by ryantusi** [comment]
Can you elaborate further or show the repository you are working with so I can help you out with what's wrong? 

The `Error 404` code means `Page not found` and the possible reasons might be probably:
1. You are not linking your image properly
2. You are not running the server on your HTML properly
3. You are using wrong image extension or referring to a wrong image
4. Your directories is in wrong order

There can be much more. Make sure to post your code here or give the link to your repository where the code is being stored.

**Comment by dhruv8433** [comment]
@Cawaale256 

Getting a 404 error when trying to display an image on your web page usually means that the path to the image file is incorrect or the image file isn't properly uploaded to GitHub. Here’s how you can ensure your images are correctly uploaded and referenced in your HTML:

## Steps to Upload Images and Reference Them Correctly
### Ensure the Image is Uploaded to GitHub:

Go to your GitHub repository.
Navigate to the folder where you want to upload the image.
Click on the "Add file" button and choose "Upload files".
Select your image file and upload it.
Check the Image Path:

### After uploading, navigate to the image in the repository and click on it.
Click on the “Download” button and copy the URL of the image.
Reference the Image in Your HTML:

### Use the copied URL in the src attribute of the <img> tag.

Example:

`<img src="https://github.com/your-username/your-repo-name/blob/main/path-to-your-image/image.jpg" alt="Description of Image">
`

**Comment by Cawaale256** [comment]
Hi there,
Thank you  Davide Presti and Soni Dhruv and I am sorry for not
responding sooner and the issue as you said to do with link in the html and
once I wrote <link rel="stylesheet" href="assets/css/style.css"
type="text/css" />
<img src="assets/img/image.jpg"> the image appear in the page

**Reply by davevad93** [reply]
You're welcome @Cawaale256 , glad you've solved it!
